docs/python/generators/examples_helpers.py

```
@@ -86,6 +86,118 @@ def _find_unquoted_word_on_nontrivial_line(word: str, text: str) -> bool:
+    lines = text.splitlines()
+    # direct name match (non-quoted preferred)
+    for name in (names or []):
+        if not name:
+            continue
+        for m in re.finditer(r"\b" + re.escape(name) + r"\b", text):
+            if _is_in_double_quotes(text, m.start()):
+                continue
+            pos = m.start()
+            cum = 0
+            line_idx = 0
+            for i, l in enumerate(lines):
```

Ruff (0.15.21)

[error] Ambiguous variable name: `l`

(E741)